server/src/test/java/org/opensearch/index/translog/transfer/BaseTranslogTransferManagerTests.java

```diff
@@ -453,29 +466,66 @@ public void testDownloadTranslog() throws IOException {
         assertTlogCkpDownloadStats();
     }
 
+    public void mockResponseDownloadBlobWithMetadata_WithNULLMetadata() throws IOException {
+        when(transferService.downloadBlobWithMetadata(any(BlobPath.class), eq("translog-23.tlog"))).thenAnswer(invocation -> {
+            Thread.sleep(delayForBlobDownload);
+            return new InputStreamWithMetadata(new ByteArrayInputStream(tlogBytes), null);
+        });
+    }
+
+    public void testDownloadTranslog_When_CkpFileStoredAsMetadata() throws IOException {
+        Path location = createTempDir();
+        mockResponseDownloadBlobWithMetadata_WithCkpFileStoredAsMetadata();
+        assertFalse(Files.exists(location.resolve("translog-23.tlog")));
+        assertFalse(Files.exists(location.resolve("translog-23.ckp")));
+        translogTransferManager.downloadTranslog("12", "23", location);
+        assertTrue(Files.exists(location.resolve("translog-23.tlog")));
+        assertTrue(Files.exists(location.resolve("translog-23.ckp")));
+        assertTlogCkpDownloadStats_when_CkpFileStoredAsMetadata();
+    }
+
+    public void mockResponseDownloadBlobWithMetadata_WithCkpFileStoredAsMetadata() throws IOException {
+        Map<String, String> metadata = TranslogCheckpointSnapshot.createMetadata(ckpBytes);
+        when(transferService.downloadBlobWithMetadata(any(BlobPath.class), eq("translog-23.tlog"))).thenAnswer(invocation -> {
+            Thread.sleep(delayForBlobDownload);
+            return new InputStreamWithMetadata(new ByteArrayInputStream(tlogBytes), metadata);
+        });
+    }
+
     public void testDownloadTranslogAlreadyExists() throws IOException {
         Path location = createTempDir();
         Files.createFile(location.resolve("translog-23.tlog"));
         Files.createFile(location.resolve("translog-23.ckp"));
-
+        mockResponseDownloadBlobWithMetadata_WithNULLMetadata();
         translogTransferManager.downloadTranslog("12", "23", location);
-
-        verify(transferService).downloadBlob(any(BlobPath.class), eq("translog-23.tlog"));
+        verify(transferService).downloadBlobWithMetadata(any(BlobPath.class), eq("translog-23.tlog"));
```

Comment:
Why should `downloadBlobWithMetadata` be getting called here in the without metadata usecase?

Reply:
Yes...As during download we won't know if the ckp is stored as metadata or stored separatly. We will always call `downloadBlobWithMetadata`